packages/blaze/builtins.js

```diff
@@ -103,9 +173,25 @@ Blaze.Each = function (argFunc, contentFunc, elseFunc) {
     }, {
       addedAt: function (id, item, index) {
         Tracker.nonreactive(function () {
-          var newItemView = Blaze.With(item, eachView.contentFunc);
+          var newDataContext;
+          if (eachView.variableName) {
```

Comment:
Maybe we can make it more explicit that this is more like `if (this is a new each)`?